main.rb

```diff
@@ -0,0 +1,121 @@
+# rubocop: disable Metrics/CyclomaticComplexity, Metrics/MethodLength,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |n|
+      yield to_a[n]
+    end
+    to_a
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |i|
+      yield to_a[i], i
+    end
+    to_a
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    result = 0
+    while result < size
+      return false unless yield(self[result])
+
+      result += 1
+    end
+
+    true
+  end
+
+  def my_any?(*_arg)
+    return true unless block_given?
+
+    result = 0
+    while result < size
+      return true if yield(self[result])
+
+      result += 1
+    end
+
+    false
+  end
```

Comment:
### #my_any 
enumerable my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
``` ruby
irb>arr = [false, false]
irb>arr.my_any?
  => true
```
> expected to be false
------
enumerable my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
``` ruby
irb>arr = %w[testdoor roadtest test]
irb>arr.my_any?(Integer)
  => true
```
> expected to be false
------
enumerable my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
``` ruby
irb>arr = %w[testdoor roadtest test]
irb>arr.my_any?(/z/)
  => true
```
> expected to be false
------
enumerable my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
``` ruby
irb>arr = %w[testdoor roadtest test]
irb>arr.my_any?('tim')
  => true
```
> expected to be false